config/typescript.js

```diff
@@ -1,20 +1,79 @@
 module.exports = {
   parser: '@typescript-eslint/parser',
+  parserOptions: {
+    'sourceType': 'module',
+  },
   plugins: [
     '@typescript-eslint',
   ],
-  extends: [
-    'eslint:recommended',
-    'plugin:@typescript-eslint/eslint-recommended',
-    'plugin:@typescript-eslint/recommended',
-  ],
   rules: {
+    // Checked by TypeScript - ts(2378)
+    'getter-return': 'off',
+    // Checked by TypeScript - ts(2300)
+    'no-dupe-args': 'off',
+    // Checked by TypeScript - ts(1117)
+    'no-dupe-keys': 'off',
+    // Checked by TypeScript - ts(7027)
+    'no-unreachable': 'off',
+    // Checked by TypeScript - ts(2367)
+    'valid-typeof': 'off',
+    // Checked by TypeScript - ts(2588)
+    'no-const-assign': 'off',
+    // Checked by TypeScript - ts(2588)
+    'no-new-symbol': 'off',
+    // Checked by TypeScript - ts(2376)
+    'no-this-before-super': 'off',
+    // Checked by TypeScript - `strictNullChecks`
+    'no-undef': 'off',
+    // Checked by TypeScript
+    'no-dupe-class-members': 'off',
+    // Checked by TypeScript
+    'no-redeclare': 'off',
+    // Checked by TypeScript
+    'import/no-unresolved': 'off',
+
     // Rules from @typescript-eslint/eslint-plugin
+    '@typescript-eslint/adjacent-overload-signatures': 'error',
     '@typescript-eslint/array-type': 'error',
-    '@typescript-eslint/consistent-type-definitions': ['error', 'type'],
+    '@typescript-eslint/ban-ts-ignore': 'error',
+    '@typescript-eslint/ban-types': 'error',
+    '@typescript-eslint/class-name-casing': 'error',
+    '@typescript-eslint/consistent-type-assertions': 'error',
+    '@typescript-eslint/consistent-type-definitions': ['error', 'interface'],
+    '@typescript-eslint/explicit-function-return-type': 'off',
```

Comment:
This rule seems to be recommended. Any reason it was left off?

Reply:
It's incredibly noisy and type inference + `noImplictAny` is a decent enough combo for now